### DISCUSSION TITLE
[Bug]: T114 GPS won't update

### DISCUSSION THREAD

**Discussion by aarondong2005** [meshtastic/firmware#5624] [Troubleshooting]
### Category

Other

### Hardware

Heltec Mesh Node T114

### Firmware Version

2.5.15

### Description

I have Heltec V3 and T114 v2. With same setting the V3 will update without issue for the route. But the T114 only update once it get the signal and no update at all. BTW, I tried all the versions early than 2.5.15 as well.

**Comment by fifieldt** [comment]
Hi, could you define "won't update" in a bit more detail?

Also, is it possible to provide log files, such as those from the serial console at flasher.meshtastic.org

**Comment by aarondong2005** [comment]
I am so sorry. I mean the GPS position won't get update from T114 v2. It will only show the position when it get the signal then no update at all.

**Comment by fifieldt** [comment]
No need to apologize :) Also, thank you for testing the latest firmware, we appreciate your support :)  

I have a T114, so I will try to confirm the problem.

To be clear, is this what happens: 
1. T114 gets a position the first time
2.  Heltec V3, and the app, can see the position of the T114
3. T114 never updates position again
?

**Reply by aarondong2005** [reply]
1. Correct
2. Yes, if I connect to the V3 and able to se T114 v2. But only one position available
3. Yes, it only show one update then no update at all.

**Comment by b8b8** [comment]
How far of a distance did you move between tests?
What app are you using to see the new location packets?

Are you testing the Heltec and T114 right next to each other? If so, the constant transmission of the Heltec (when driving) may be pushing the channel utilization higher, causing the T114 to self limit its position packet interval. when connected to the T114 do you see the Lat/Lon change but just not transmitted and show on another Meshtastic map?

What are your actual position settings for the T114 (update inverval, smart position etc.)?

Edit:
I can do some testing with my T114 today.

Initial testing with firmware 2.5.16 shows i am getting position updates
![image](https://github.com/user-attachments/assets/261d577d-799c-44a5-8494-8b3ebe146305)

**Reply by aarondong2005** [reply]
Today I tested it again. This time I didn't bring Heltec V3 with me, I just used the T114 v2. I went from Ontario to Irvine drive about 30 miles. It only show one update. I attached my position setting for the T114 v2. 
![IMG_9112](https://github.com/user-attachments/assets/72e4321e-e20d-4363-b94b-edee6a8c5d68)
![IMG_9113](https://github.com/user-attachments/assets/8e0e2bf0-ffc0-4893-9719-e7d7a82d1fd7)
![IMG_9125](https://github.com/user-attachments/assets/fbfac5f8-a489-460c-847d-fd7846ec01a9)

**Reply by fifieldt** [reply]
May I confirm which role your device is set to? client? Router?

**Reply by aarondong2005** [reply]
I set to client. I also tried tracker. Both only GPS update once

**Reply by b8b8** [reply]
Maybe somewhat related, but i would set your GPS update interval and smart position interval to the same 30 seconds. But also i found on a different device (SEEED T1000E) the GPS doesnt really like an updated interval quicker than 60 seconds and causes a hang, so i would actually change both of those to 60 seconds and retest. Mine are set to 120 seconds. 

Also what case are you using and where is the device placed in your car when driving? 
This is my setup, and we know from previous testing years ago the 15mmx15mm GPS antennas are affected by the thickness (especially when indoors) of the cases. Though you seem to be getting a satellite lock so is probably not the case for you. 
![image](https://github.com/user-attachments/assets/f688b50a-30fb-44b4-b2ff-fdd46c0ffcab)
![image](https://github.com/user-attachments/assets/55c35044-fa86-43e1-b348-495978b7cbe8)

**Reply by aarondong2005** [reply]
I just find out the position on the T114 is correct. But the phone app will only update once and it is incorrect. I will check to see if the T114 device  GPS position will change or not

**Reply by b8b8** [reply]
awesome, It might be a background location setting from your phone thats the problem then. For testing i would just turn that off completely and not share the phone location with the node. try and complete the tests with purely the T114 hardware GPS.

**Reply by aarondong2005** [reply]
I tested on 2 of the T114 v2. T114 screen will update the GPS position if I move and drive but the phone app will not (both iPhone and Android)

**Reply by b8b8** [reply]
This might be an app settings problem then. Do you allow the app to always run in the background? Is the bluetooth going to sleep?

What phones and OS versions are your running the tests with?

Post pictures of your position settings, and app permissions.

**Reply by aarondong2005** [reply]
You should not thinking about app setting related issue. Because I have no issue with Heltec V3. So it is not phone app issue. The issue should be either T114 v4 not upload the newer GPS position. Or the phone app not trying to get the newer GPS position form T114

**Reply by b8b8** [reply]
ok but we are still trying to narrow it down.
Maybe its just NRF devices having bluetooth issues. today my T1000E was having a GPS issue passing it on to the android app.
What phones and OS versions are you using and do you have any power savings enabled?

https://github.com/meshtastic/firmware/issues/5656#issuecomment-2560184915

**Reply by aarondong2005** [reply]
The test phone is iPhone 15 Pro Max + iOS 18.2 + Meshtastic 2.5.14 (1327). No Power savings enabled
I Just reset the T114 NodeDB and did fastory reset. Let's see the difference. BTW, I also reinstalled the app as well

**Reply by b8b8** [reply]
i tested with my iPad and it seems to be working for me. Same as you 18.2, though 2.5.14 (132**6**)

**Reply by aarondong2005** [reply]
Still the same result. Only got one update then no more from the app......

**Reply by b8b8** [reply]
hmmm. I just installed a T114 in my car solely for tracking and to act as a repeater and it performed well all weekend. What are your channel settings (public or private) and what order are they?

**Reply by garthvh** [reply]
If you are on the default channel with reduced precision or the node does not move there is no history stored, it the timestamp updating?

**Reply by aarondong2005** [reply]
> hmmm. I just installed a T114 in my car solely for tracking and to act as a repeater and it performed well all weekend. What are your channel settings (public or private) and what order are they?

My issue still remains even after I flash to v2.5.19. Also I tried the Reset NodeBD and Factory Reset. Nothing will make the GPS update more than just one. Are you part of the Meshtastic team? Can I send my device to you to take test?

**Reply by b8b8** [reply]
I am not part of the Meshtastic team. Please post every page of your channel settings and position settings.

**Reply by b8b8** [reply]
> Still the same result. Only got one update then no more from the app......

What channels do you have loaded and in what order?

**Comment by b8b8** [comment]
i continued testing all day, and have a position packets consistently throughout. I have ended the test as i have not experienced any issues (even when having the node sitting at home for a few hours with smart position enabled i got a few stationary updates).

**Reply by aarondong2005** [reply]
Are you using the T114 original L76K GPS module?

**Reply by b8b8** [reply]
Yes.

**Comment by mcohen333** [comment]
pls advise i try to use heltec t114 rev 2 with gps l76k but no connection to the gps does any one can help with simple code that read the gps and put the data n serial port
regards

**Comment by aarondong2005** [comment]
Finally I was able to find the issue and fixed the Heltec T114 v2 GPS no update. It is related to the iOS and Android channel setting. 

iPhone is my main device. I have a Android phone but never use it for the Meshtastic. Someone mentioned "imprecise location" in the channel setting on Reddit. So I go over the iOS app and not able to find it in "LongFast" default channel. But on the Android app, there is "Precise location" and it is off. After I turn it on the GPS start to update as Hecltec V3. I think it is bug for the iOS app without having this setting.

App version used:
1. iOS - 2.5.17 (1347)
2. Android - 2.5.16

Solution:
1. On the iOS there is no "Precise location" setting. Even changed "Approximate Location" from 1.8 miles to 0.9 miles won't make and change. Also the GPS location on the map is very big circle

![IMG_9542](https://github.com/user-attachments/assets/4a906404-f2ce-4a95-aac0-9ec9a0dfdf00)
![IMG_9541](https://github.com/user-attachments/assets/78081ecc-3deb-4c1a-81a9-77ddefa19b06)
![IMG_9543](https://github.com/user-attachments/assets/f69b6724-1cd4-4e7e-a57d-1fa087ecddf1)

2. On the Android. go to the default "LongFast" setting. You will able to see the "Precise location" and it is Off by default. Change it to On.

![Screenshot_2025-02-02-11-58-12-934_com geeksville mesh](https://github.com/user-attachments/assets/01c483c0-f18c-4bbe-a113-02215e944d80)
![Screenshot_2025-02-02-11-58-17-289_com geeksville mesh](https://github.com/user-attachments/assets/920a7f89-4e5c-4a03-a7cd-73b0732ec122)

3. The GPS position circle on the map become much smaller and the position will start to update
![IMG_9539](https://github.com/user-attachments/assets/58bfb87b-5860-4c0d-9a83-2cfd06d12ae4)

**Reply by b8b8** [reply]
Yup, that's why both Garth and I asked about your channel settings. iOS/app store have different privacy requirements for the default channel than android. 

It is not a bug for iOS. If you would still like precise location on your iphone, you can override the safety feature with an android device, or the preferred route is to set your primary channel as private with longfast as a secondary public channel. This will allow the precise setting to appear for your private channel to share location with your friends.

**Comment by fifieldt** [comment]
Congratulations on finding the issue. Thank you for your patience and sharing your findings.
